src/main/java/com/mineboundteam/minebound/entity/MyrialSwordEntity.java

```diff
@@ -0,0 +1,93 @@
+package com.mineboundteam.minebound.entity;
+
+import com.mineboundteam.minebound.entity.registry.EntityRegistry;
+import com.mineboundteam.minebound.item.registry.ItemRegistry;
+import it.unimi.dsi.fastutil.ints.IntOpenHashSet;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.projectile.ProjectileUtil;
+import net.minecraft.world.entity.projectile.ThrowableItemProjectile;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.EntityHitResult;
+import net.minecraft.world.phys.HitResult;
+import net.minecraft.world.phys.Vec3;
+import org.jetbrains.annotations.NotNull;
+
+public class MyrialSwordEntity extends ThrowableItemProjectile {
```

Comment:
Should not pass through blocks

Reply:
Now it does not stay out when it collides with blocks; should hover at that location when it hits a block